app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,79 @@
+class DriversController < ApplicationController
+
+  def index
+    @drivers = Driver.all
+  end
+
+  def show
+    driver_id = params[:id].to_i
+    @driver = Driver.find_by(id: driver_id)
+
+    if @driver.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def new
+    @driver = Driver.new
+  end
+
+  def create
+    @driver = Driver.new(driver_params)
+
+    if @driver.save
+      redirect_to driver_path(@driver)
+      return
+    else
+      render :new, status: :bad_request
+      return
+    end
+  end
+
+  def find_by_id
+    driver_id = params[:id].to_i
+    driver = Driver.find_by(id: driver_id)
+  end
```

Comment:
This might have made more sense as a method on `Driver`.  

DRYing this sort of thing up is what controller filters are for though so good instincts!